### PR TITLE
Add ability to create intermediate CA cert

### PR DESCRIPTION
**Pull request by csstaub**
r: @alokmenghrajani

### CODE REVIEW

**Comment by alokmenghrajani** [comment]
👍

File: pkix/cert_auth.go [hunk 1]
```diff
@@ -110,3 +110,51 @@ func CreateCertificateAuthority(key *Key, organizationalUnit string, years int,
...
+	}
+	authTemplate.SerialNumber.Set(serialNumber)
+	authTemplate.MaxPathLenZero = false
+
+	rawCsr, err := csr.GetRawCertificateSigningRequest()
+	if err != nil {
+		return nil, err
+	}
+
+	authTemplate.Subject = rawCsr.Subject
```
Comment (csstaub) [inline pkix/cert_auth.go]: This should probably be `RawSubject`

File: pkix/cert_auth.go [hunk 2]
```diff
@@ -110,3 +110,51 @@ func CreateCertificateAuthority(key *Key, organizationalUnit string, years int,
...
+		return nil, err
+	}
+
+	authTemplate.RawSubject = rawCsr.RawSubject
+
+	caExpiry := time.Now().Add(crtAuth.GetExpirationDuration())
+	proposedExpiry := time.Now().AddDate(years, 0, 0).UTC()
+	// ensure cert doesn't expire after issuer
+	if caExpiry.Before(proposedExpiry) {
+		authTemplate.NotAfter = caExpiry
```
Comment (alokmenghrajani) [inline pkix/cert_auth.go]: we should show a warning when we take this branch.

**Comment by alokmenghrajani** [comment]
lgtm for reals this time :)

File: pkix/cert_auth.go (hunk 2 above)
Comment (csstaub) [inline pkix/cert_auth.go]: yeah. probably ditto for host certs.

**Pull request merged**